src/integration/java/org/humancellatlas/ingest/biomaterial/BiomaterialControllerTest.java

```diff
@@ -0,0 +1,222 @@
+        Process process2 = new Process();
+        processRepository.save(process2);
+
+        webApp.perform(post("/biomaterials/{id}/derivedByProcesses/", biomaterial.getId())
+                .contentType("text/uri-list")
+                .content(ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process.getId()
+                        + '\n' + ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process2.getId()))
+                .andExpect(status().isOk());
+
+        verifyInDraft(biomaterial, process, process2);
+
+        Biomaterial updatedBiomaterial = biomaterialRepository.findById(biomaterial.getId()).get();
+        assertThat(updatedBiomaterial.getDerivedByProcesses())
+                .usingElementComparatorOnFields("id")
+                .containsExactly(process, process2);
+    }
+
+    @Test
+    public void testOverrideLinkFileAsDerivedByMultipleProcessesDefaultPutEndpoint() throws Exception {
+        biomaterial.addAsDerivedByProcess(process);
+        biomaterialRepository.save(biomaterial);
+
+        Process process2 = new Process();
+        Process process3 = new Process();
+        processRepository.save(process2);
+        processRepository.save(process3);
+
+        webApp.perform(put("/biomaterials/{id}/derivedByProcesses/", biomaterial.getId())
+                .contentType("text/uri-list")
+                .content(ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process2.getId()
+                        + '\n' + ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process3.getId()))
+                .andExpect(status().isOk());
+
+        verifyInDraft(biomaterial, process, process2, process3);
+
+        Biomaterial updatedBiomaterial = biomaterialRepository.findById(biomaterial.getId()).get();
+        assertThat(updatedBiomaterial.getDerivedByProcesses())
+                .usingElementComparatorOnFields("id")
+                .containsExactly(process2, process3);
+    }
+
+    @Test
+    public void testOverrideDefaultDerivedByProcessesLinkEndpoint() throws Exception {
+        webApp.perform(post("/biomaterials/{id}/derivedByProcesses/", biomaterial.getId())
+                .contentType("text/uri-list")
+                .content(ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString() + "/processes/" + process.getId()))
+                .andExpect(status().isOk());
+
+        verifyInDraft(biomaterial, process);
+
+        Biomaterial updatedBiomaterial = biomaterialRepository.findById(biomaterial.getId()).get();
+        assertThat(updatedBiomaterial.getDerivedByProcesses())
+                .usingElementComparatorOnFields("id")
+                .containsExactly(process);
+    }
+
+    @Test
+    public void testUnlinkBiomaterialAsInputToProcessChangesTheirValidationStatesToDraft() throws Exception {
```

Comment:
Test case name tells less than the test does. The test also validates the `process` has been removed as an input process of the `biomaterial`.

Reply:
I renamed it to just say test method, and testing the entity is unlinked is one of the expected behaviours.